toffy/streak_detection.py

```diff
@@ -0,0 +1,429 @@
+        the intensity. Defaults to 95.
+
+    Returns:
+        np.ndarray: The binary mask containing all of the candidate strokes.
+    """
+    if channel is not None:
+        x: np.ndarray = channel
+        # Denoise the Image
+        x = restoration.denoise_wavelet(
+            x, wavelet="db2", mode="hard", rescale_sigma=True
+        )
+        # Rescale the intensity using percentile ranges
+        pmin_v, pmax_v = np.percentile(x, (pmin, pmax))
+        x = exposure.rescale_intensity(x, in_range=(pmin_v, pmax_v))
+        # Laplace filter to get the streaks
+        x = filters.laplace(x, ksize=3)
+        x = exposure.rescale_intensity(x, out_range=(0, 1))
+        # Smoothing
+        x = filters.gaussian(x, sigma=(0, gaussian_sigma))  # (y, x)
+        # Exposure Adjustments
+        x = exposure.adjust_gamma(x, gamma=gamma, gain=gamma_gain)
+        x = exposure.adjust_log(x, gain=log_gain, inv=True)
+        x = exposure.rescale_intensity(x, out_range=(0, 1))
+        # apply threshold
+        x = x > 0.3
+
+        return x
+
+
+def _create_mask(streak_data: StreakData, channel: np.ndarray) -> None:
+    """Creates the initial streak mask which is then used for filtering and labeling appropriate
+    streaks.
+
+    Args:
+        streak_data (StreakData): An instance of the StreakData Dataclass, holds all necessary
+        data for streak correction.
+        channel (np.ndarray): The input channel (Numpy array representation of the tiff file) to
+        be used for streak detection.
+    """
+    streak_data.streak_mask = _binary_mask(channel=channel, pmin=2, pmax=98)
+
+
+def _filter_mask(streak_data: StreakData) -> None:
+    """Filters the streaks. Removes incorrectly masked streaks which are too short and do not have
+    a high eccentricity.
+
+    Args:
+        streak_data (StreakData): An instance of the StreakData Dataclass, holds all necessary
+        data for streak correction.
+    """
+    # Label all the candidate streaks
+    labeled_streaks = measure.label(
+        streak_data.streak_mask, connectivity=2, return_num=False
+    )
+    # Gather properties of all the candidate streaks.
+    streak_data.streak_df = pd.DataFrame(
+        measure.regionprops_table(
+            label_image=labeled_streaks,
+            cache=True,
+            properties=[
+                "label",
+                "bbox",
+                "eccentricity",
+                "area",
+            ],
+        )
+    ).rename(
+        {
+            "bbox-0": "min_row",
+            "bbox-1": "min_col",
+            "bbox-2": "max_row",
+            "bbox-3": "max_col",
+        },
+        axis="columns",
+    )
+    streak_data.streak_df.index.names = ["index"]
+
+    # Filter out eccentricities that are less than 0.99999 (only keep straight lines)
+    # Filter out small areas (small lines)
+    eccentricity_value = 0.99999
+    area_value = 50
```

Comment:
area should be a modifiable parameter, at least for now, until we see what needs to be changed when